src/main/java/com/devcycle/sdk/server/common/model/EvalHooksRunner.java

```diff
@@ -0,0 +1,119 @@
+package com.devcycle.sdk.server.common.model;
+
+import com.devcycle.sdk.server.common.exception.AfterHookError;
+import com.devcycle.sdk.server.common.exception.BeforeHookError;
+import com.devcycle.sdk.server.common.logging.DevCycleLogger;
+
+import javax.swing.text.html.Option;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A class that manages evaluation hooks for the DevCycle SDK.
+ * Provides functionality to add and clear hooks, storing them in an array.
+ */
+public class EvalHooksRunner {
+    private List<EvalHook> hooks;
+
+    /**
+     * Default constructor initializes an empty list of hooks.
+     */
+    public EvalHooksRunner() {
+        this.hooks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a single hook to the collection.
+     * 
+     * @param hook The hook to add
+     */
+    public void addHook(EvalHook hook) {
+        if (hook != null) {
+            hooks.add(hook);
+        }
+    }
+
+    /**
+     * Clears all hooks from the collection.
+     */
+    public void clearHooks() {
+        hooks.clear();
+    }
+
+    /**
+     * Runs all before hooks in order.
+     *
+     * @param context The context to pass to the hooks
+     * @param <T> The type of the variable value
+     * @return The potentially modified context
+     */
+    public <T> HookContext<T> executeBefore(HookContext<T> context) {
+        HookContext<T> beforeContext = context;
+        for (EvalHook<T> hook : hooks) {
+            try {
+                Optional<HookContext<T>> newContext = hook.before(beforeContext);
+                if (newContext.isPresent()) {
+                    beforeContext = beforeContext.merge(newContext.get());
+                }
+            } catch (Exception e) {
+                throw new BeforeHookError("Before hook failed", e);
+            }
+        }
+        return beforeContext;
+    }
+
+    /**
+     * Runs all after hooks in reverse order.
+     *
+     * @param context The context to pass to the hooks
+     * @param variable The variable result to pass to the hooks
+     * @param <T> The type of the variable value
+     */
+    public <T> void executeAfter(HookContext<T> context, Variable<T> variable) {
+        for (EvalHook<T> hook : hooks) {
```

Comment:
this will execute the hooks in the same order where we want the reverse order here